src/snakemake/resources.py

```
@@ -502,104 +409,539 @@
+    def default(cls, mode: Literal["bare"] | Literal["full"] | None = None):
+        """Return an object loaded with a standard default resource set.
+
+        Arguments
+        ========
+        mode: "bare" or "full"
+            full presets "tmpdir", "mem_mb", and "disk_mb". bare sets only "tmpdir"
+        """
+        try:
+            default_resources = cls.DEFAULTS[mode or "full"]
+        except KeyError:
+            raise ValueError(f"Unexpected mode for DefaultResources: {mode}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)